tools/SetupFlow/DevHome.SetupFlow/ViewModels/LoadingViewModel.cs

```diff
@@ -377,73 +373,109 @@ private void SetExecutingTaskAndActionCenter()
+    /// <param name="information">Used to know if the computer needs to reboot.</param>
+    /// <param name="finishedState">The state of the finished task.</param>
+    /// <returns>A LoadingMessageViewModel that can be placed into the UI.</returns>
+    private LoadingMessageViewModel GenerateFinishedMessage(TaskInformation information, TaskFinishedState finishedState)
     {
-        _log.Debug($"Updating message for task {information.MessageToShow} with state {taskFinishedState}");
+        _log.Debug($"Updating message for task {information.MessageToShow} with state {finishedState}");
         var stringToReplace = string.Empty;
         BitmapImage statusSymbolIcon = null;
 
         // Two things to do.
         // 1. Change the message color and icon in information
         // 2. Add a new message with the done message.
-        if (taskFinishedState == TaskFinishedState.Success)
+        if (finishedState == TaskFinishedState.Success)
         {
             if (information.TaskToExecute.RequiresReboot)
             {
-                _log.Debug("Task succeeded but requires reboot; adding to action center");
                 stringToReplace = information.TaskToExecute.GetLoadingMessages().NeedsReboot;
                 statusSymbolIcon = (_currentTheme == ElementTheme.Dark) ? DarkCaution : LightCaution;
-                ActionCenterItems.Insert(0, information.TaskToExecute.GetRebootMessage());
             }
             else
             {
-                _log.Debug("Task succeeded");
                 stringToReplace = information.TaskToExecute.GetLoadingMessages().Finished;
                 statusSymbolIcon = (_currentTheme == ElementTheme.Dark) ? DarkSuccess : LightSuccess;
             }
+        }
+        else if (finishedState == TaskFinishedState.Failure)
+        {
+            stringToReplace = information.TaskToExecute.GetLoadingMessages().Error;
+            statusSymbolIcon = (_currentTheme == ElementTheme.Dark) ? DarkError : LightError;
+        }
+
+        var newLoadingScreenMessage = _host.GetService<LoadingMessageViewModel>();
+        newLoadingScreenMessage.MessageToShow = stringToReplace;
+        newLoadingScreenMessage.StatusSymbolIcon = statusSymbolIcon;
+
+        return newLoadingScreenMessage;
+    }
+
+    /// <summary>
+    /// Updates the loading screen task logging UI to show that the task is finished.
+    /// </summary>
+    /// <param name="originalMessage">The executing message.</param>
+    /// <param name="finishedMessage">The finished message.</param>
+    private void InsertFinishedMessageIntoLogScreen(LoadingMessageViewModel originalMessage, LoadingMessageViewModel finishedMessage)
+    {
+        // Remove the executing message from the list.
+        ExecutingMessages.Remove(originalMessage);
+
+        // Insert the same message.  All messages in this list have their foreground set to

```

Comment:
Do we even need this? Seems like just bloating the output. We get pairs of messages like:
Cloning 68K
Couldn't clone 68K

"Cloning 68K" is redundint now as it doesn't have any additional info comparing to the finishedMessage.

Reply:
The first iteration replaced the messages.  The designers wanted a scrolling log style.  They wanted to keep the old messages.